addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+    # False => Not sent yet.
+    l10n_tw_edi_state = fields.Selection(
+        selection=[("invoiced", "Invoiced"), ("valid", "Valid"), ("invalid", "Invalid")],
+        string="Invoice Status",
+        copy=False,
+        readonly=True,
+        tracking=True,
+    )
+    l10n_tw_edi_ecpay_is_included_tax = fields.Selection(
+        selection=[("1", "tax included"), ("0", "untax")],
+        string="Included Tax",
+        compute="_compute_is_included_tax",
+    )
+    l10n_tw_edi_is_donate = fields.Boolean(string="Donate", copy=False, compute="_compute_is_donate", store=True, readonly=False)
+    l10n_tw_edi_love_code = fields.Char(string="Love Code", copy=False, compute="_compute_is_donate", store=True, readonly=False)
+    l10n_tw_edi_is_print = fields.Boolean(string="Get Printed Version", copy=False, compute="_compute_is_print", store=True, readonly=False)
+    l10n_tw_edi_carrier_type = fields.Selection(
+        selection=[("1", "ECpay e-invoice carrier"), ("2", "Citizen Digital Certificate"), ("3", "Mobile Barcode")],
+        string="Carrier Type",
+        copy=False,
+        compute="_compute_carrier_info",
+        store=True,
+        readonly=False)
+    l10n_tw_edi_carrier_number = fields.Char(string="Carrier Number", copy=False, compute="_compute_carrier_info", store=True, readonly=False)
+    l10n_tw_edi_invoice_amount = fields.Float(string="Invoice Amount", readonly=True, copy=False)
+    l10n_tw_edi_invoice_shop_custom_number = fields.Char(string="Cooperative Shop Custom Number", readonly=True, copy=False)
+    l10n_tw_edi_invoice_create_date = fields.Datetime(string="Creation Date", readonly=True, copy=False)
+    l10n_tw_edi_refund_finish = fields.Boolean(string="Refund Finish", readonly=True, copy=False)
+    l10n_tw_edi_refund_state = fields.Selection(
+        selection=[
+            ("draft", "draft"),
+            ("to be agreed", "To be agreed"),
+            ("agreed", "Agreed"),
+            ("disagree", "Disagree"),
+        ],
+        string="Refund State",
+        default="draft",
+        readonly=True,
+        copy=False,
+    )
+    l10n_tw_edi_refund_agreement_type = fields.Selection(
+        [("offline", "Offline Agreement"), ("online", "Online Agreement")],
+        default="offline",
+        string="Refund invoice Agreement Type",
+        required=True,
+        copy=False,
+    )
+    l10n_tw_edi_origin_invoice_number = fields.Many2one('account.move', string="Original Invoice", readonly=True, copy=False)
+    l10n_tw_edi_refund_invoice_number = fields.Char(string="Refund Invoice Number", readonly=True, copy=False)
+    l10n_tw_edi_remain_refundable_amount = fields.Float(string="Remain refundable amount", readonly=True, copy=False)
+
+    @api.depends('l10n_tw_edi_is_donate', 'l10n_tw_edi_carrier_type')
+    def _compute_is_print(self):
+        for move in self:
+            if move.l10n_tw_edi_is_donate or move.l10n_tw_edi_carrier_type:
```

Comment:
```suggestion
            if move.l10n_tw_edi_is_donate or (self.partner_id.vat and move.l10n_tw_edi_carrier_type in [1, 2]):
```
I think this is more correct following the doc

Reply:
![image](https://github.com/user-attachments/assets/6f72456d-7291-415f-822e-86dcacfa1cdd)
![image](https://github.com/user-attachments/assets/28feb09e-cb11-4dc7-9bdd-0d0de5886531)
The doc is different from what I tested.
If you selected Print, you cannot choose any Carrier type, the API returns error

I can make the change, it will return error anyway while calling the API